src/Catalog/Persistence/Storage.cs

```diff
@@ -191,8 +205,23 @@ protected void TraceMethod(string method, Uri resourceUri)
         {
             if (Verbose)
             {
-                Trace.WriteLine(String.Format("{0} {1}", method, resourceUri));
+                //The Uri depends on the storage implementation.
+                Uri storageUri = GetUri(GetName(resourceUri));
+                Trace.WriteLine(String.Format("{0} {1}", method, storageUri));
             }
         }
+
+        private string TraceException(string method, Uri resourceUri, Exception exception)
+        {
+            string message = $"{method} EXCEPTION: {GetUri(GetName(resourceUri))} {exception.Message}";
```

Comment:
You should log the full exception instead of just its `Message` so that `InnerException`s are logged too.

Reply:
Updated.